### DISCUSSION TITLE
Tab Behavior, Search Bar Error, and Sidebar Configuration in Fumadocs UI

### DISCUSSION THREAD

**Discussion by Shubham-EV** [fuma-nama/fumadocs#495] [Docs Feedback]
## Issues

1. When I click on different tab items, other tabs with the same name change. When I click on one, the others should not change.

https://github.com/fuma-nama/fumadocs/assets/64977436/bda0b788-3299-4262-a1d3-983d9d192db4

2. When I cloned this repository [fumadocs-ui-template](https://github.com/fuma-nama/fumadocs-ui-template) and ran `npm i` I got this error on clicking search bar

https://github.com/fuma-nama/fumadocs/assets/64977436/a07c7e4b-a00a-43a2-884a-cbc41af91038

3. I am not able to understand how to configure the sidebar exactly. For example, consider this template: [fumadocs-ui-template](https://github.com/fuma-nama/fumadocs-ui-template). What code should I modify and how?and also I find the hover sidebar animation laggy. How can I disable it?

https://github.com/fuma-nama/fumadocs/assets/64977436/a0835aa9-e90f-48dd-9315-0fe8489d75dd

The same issue applies to other components. What code should I modify or create, and how should I configure them?
For eg - nextra have `theme.config.tsx` where I can modify or create custom components . I am not able to understand how to create or modify components  in fumadocs.

You have indeed provided code for configuration of component but where should I put them?
![image](https://github.com/fuma-nama/fumadocs/assets/64977436/7d6d4c7b-4f3e-412b-bcae-bf06a4b6f306)

In which file I should put it?

![image](https://github.com/fuma-nama/fumadocs/assets/64977436/7bf088ec-c524-4c99-bf85-f68f30020c71)

4. This feature is also absent in fumadocs but present in nextra.

https://github.com/fuma-nama/fumadocs/assets/64977436/0a8a8dfd-af7e-4275-a0f5-46b71f8c1ced

and also when I type ```java .  It don't give logo for java but for other programming language like js, cpp I get logo in syntax highlighting.

**Comment by fuma-nama** [comment]
1. This is a feature called shared value. It is enabled there, so that after choosing a content source, users don't have to switch the other tabs again to see examples for their content source. You can check the docs.
2. Cannot reproduce this problem on CodeSandbox, please provide a minimal reproducible repo.
3. You can read the docs before opening a question: https://fumadocs.vercel.app/docs/ui/blocks/layout, it is also answered in the FAQ section of Getting Started page. A simple walk through on the files of template should find `docs/layout.tsx` for docs layout.
4. Java icon is not allowed: https://github.com/simple-icons/simple-icons/issues/7374. This is reported by Oracle so I'm not planning to add it

**Comment by Shubham-EV** [comment]
1. Thanks
2. I think I am getting the error in `@types/node:20.14.6` and search bar works fine on `@types/node:20.14.5`
3. How to disable that hover sidebar open animation only? I want to keep collapsible: `true` but disable that animation?. Please add another property to disable it as well.
4. Thanks (but what about that code expandable button? I uploaded a video about it too. It is present in `nextra`

and is their any way to change placeholder text of serch bar?

**Comment by fuma-nama** [comment]
I'm not planning to add the expend button to code blocks, it feels a bit useless and I want to focus on something else. I'll recommend to disable the collapsible sidebar if the animation caused some problems

**Comment by Shubham-EV** [comment]
Are you planning to add a `banner`? It is useful for announcing something like new version.

**Reply by fuma-nama** [reply]
Just add a little element with sticky position to the top of body, I think people would love a fully customisable banner since its code is quite simple.

**Reply by Shubham-EV** [reply]
Yeah, I tried creating a banner, but I'm just not able to make the state persistent after clicking the close button because I'm a newbie. It would be nice if you could consider adding this in future updates in a proper way :)

For reference, this was my `banner.tsx` 

```js
"use client";
import React, { useState } from "react";

const Banner = () => {
  const [showBanner, setShowBanner] = useState(true);

  const handleClose = () => {
    setShowBanner(false);
  };

  return (
    showBanner && (
      <div className="sticky top-0 tw h-10 text-[14px] font-semibold flex items-center px-4 bg-gray-800 text-white">
        <div className="flex-grow text-center">Banner</div>
        <button onClick={handleClose} className="focus:outline-none">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="18"
            height="18"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeLinejoin="round"
            className="lucide lucide-x text-gray-300 hover:text-white"
          >
            <path d="M18 6 6 18" />
            <path d="m6 6 12 12" />
          </svg>
        </button>
      </div>
    )
  );
};

export default Banner;
```

and I used this component in my `layout.tsx`

```js
import { RootProvider } from "fumadocs-ui/provider";
import "fumadocs-ui/style.css";
import { Inter } from "next/font/google";
import type { ReactNode } from "react";

import "fumadocs-ui/style.css";
import Banner from "./banner/banner";

const inter = Inter({
  subsets: ["latin"],
});

export default function Layout({ children }: { children: ReactNode }) {
  return (
    <html lang="en" className={inter.className} suppressHydrationWarning>
      <body>
        <RootProvider>
          <Banner />
          {children}
        </RootProvider>
      </body>
    </html>
  );
}
```

Output:-

![image](https://github.com/fuma-nama/fumadocs/assets/64977436/0570bbea-d923-441c-af90-9546004b0740)

**Reply by fuma-nama** [reply]
I see, maybe add this as a component in Fumadocs UI?

**Reply by Shubham-EV** [reply]
yes that would be great.

**Reply by fuma-nama** [reply]
Added the banner component, it avoids the flash (layout shift) and allows the state to be persisted, feel free to check it out

**Comment by fuma-nama** [comment]
> and is their any way to change placeholder text of serch bar?

Use the I18n Provider, it's a part of Fumadocs UI to replace certain translations of the labels

**Reply by Shubham-EV** [reply]
Thanks :)
